src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs

```diff
@@ -300,6 +300,7 @@ internal static bool DoSslHandshake(SafeSslHandle context, ReadOnlySpan<byte> in
                 {
                     if (sendCount <= 0)
                     {
+                        ArrayPool<byte>.Shared.Return(sendBuf);
```

Comment:
I'd feel ever so slightly better if this were instead:
```C#
byte[] tmp = sendBuf;
sendBuf = null;
ArrayPool<byte>.Shared.Return(tmp);
```
If, for example, Crypto.ErrClearError were to throw an exception, with the code as written we could have returned the buffer to the pool but the caller could still have a reference via the out argument that was already assigned.